docs_src/examples/use_cases/example_first_order_resnet.py

```diff
@@ -1,101 +0,0 @@
-    predictions = output.argmax(dim=1, keepdim=True).view_as(targets)
-    return predictions.eq(targets).float().mean().item()
-
-
-x, y = load_one_batch_mnist(batch_size=BATCH_SIZE)
-x, y = x.to(DEVICE), y.to(DEVICE)
-
-
-# %%
-# We can build a ResNet by extending :py:class:`torch.nn.Module`.
-# As long as the layers with parameters
-# (:py:class:`torch.nn.Conv2d` and :py:class:`torch.nn.Linear`) are
-# ``nn`` modules, BackPACK can extend them,
-# and this is all that is needed for first order extensions.
-# We can rewrite the forward to implement the residual connection,
-# and :py:func:`extend() <backpack.extend>` the resulting model.
-
-
-class MyFirstResNet(torch.nn.Module):
-    def __init__(self, C_in=1, C_hid=5, input_dim=(28, 28), output_dim=10):
-        super().__init__()
-
-        self.conv1 = torch.nn.Conv2d(C_in, C_hid, kernel_size=3, stride=1, padding=1)
-        self.conv2 = torch.nn.Conv2d(C_hid, C_hid, kernel_size=3, stride=1, padding=1)
-        self.linear1 = torch.nn.Linear(input_dim[0] * input_dim[1] * C_hid, output_dim)
-        if C_in == C_hid:
-            self.shortcut = torch.nn.Identity()
-        else:
-            self.shortcut = torch.nn.Conv2d(C_in, C_hid, kernel_size=1, stride=1)
-
-    def forward(self, x):
-        residual = self.shortcut(x)
-        x = self.conv2(F.relu(self.conv1(x)))
-        x = x + residual  # don't use: x += residual
-        x = x.view(x.size(0), -1)
-        x = self.linear1(x)
-        return x
-
-
-model = extend(MyFirstResNet()).to(DEVICE)
-
-# %%
-# Using :py:class:`BatchGrad <backpack.extensions.BatchGrad>` in a
-# :py:class:`with backpack(...) <backpack.backpack>` block,
-# we can access the individual gradients for each sample.
-#
-# The loss does not need to be extended in this case either, as it does not
-# have model parameters and BackPACK does not need to know about it for
-# first order extensions. This also means you can use any custom loss function.
-
-model.zero_grad()
-loss = F.cross_entropy(model(x), y, reduction="sum")
-with backpack(BatchGrad()):
-    loss.backward()
-
-print("{:<20}  {:<30} {:<30}".format("Param", "grad", "grad (batch)"))
-print("-" * 80)
-for name, p in model.named_parameters():
-    print(
-        "{:<20}: {:<30} {:<30}".format(name, str(p.grad.shape), str(p.grad_batch.shape))
-    )
-
-# %%
-# To check that everything works, let's compute one individual gradient with
-# PyTorch (using a single sample in a forward and backward pass)
-# and compare it with the one computed by BackPACK.
-
-sample_to_check = 1
-x_to_check = x[sample_to_check, :].unsqueeze(0)
-y_to_check = y[sample_to_check].unsqueeze(0)
-
-model.zero_grad()
-loss = F.cross_entropy(model(x_to_check), y_to_check)
-loss.backward()
-
-print("Do the individual gradients match?")
-for name, p in model.named_parameters():
-    match = torch.allclose(p.grad_batch[sample_to_check, :], p.grad, atol=1e-7)
-    print("{:<20} {}".format(name, match))
```

Comment:
Suggestion: Would it make sense to link to the new example to keep the web address functioning?

Reply:
Yes, makes sense.
This would be relevant if there are links in the www to this example. Is there a way to find this out?

Reply:
Googling it does not yield any results:
- [web address](https://www.google.com/search?q=https%3A%2F%2Fdocs.backpack.pt%2Fen%2Fmaster%2Fuse_cases%2Fexample_first_order_resnet.html%23sphx-glr-use-cases-example-first-order-resnet-py&client=ubuntu&hs=XXU&channel=fs&ei=2NwUYee-Co_vkgXpsoKYDQ&oq=https%3A%2F%2Fdocs.backpack.pt%2Fen%2Fmaster%2Fuse_cases%2Fexample_first_order_resnet.html%23sphx-glr-use-cases-example-first-order-resnet-py&gs_lcp=Cgdnd3Mtd2l6EAMyBwgAEEcQsAMyBwgAEEcQsAMyBwgAEEcQsAMyBwgAEEcQsAMyBwgAEEcQsAMyBwgAEEcQsAMyBwgAEEcQsAMyBwgAEEcQsANKBAhBGABQxB1Y6B5gjiNoAHAEeACAAQCIAQCSAQCYAQCgAQHIAQjAAQE&sclient=gws-wiz&ved=0ahUKEwjnisa2iavyAhWPt6QKHWmZANMQ4dUDCA0&uact=5)
- [words](https://www.google.com/search?q=backpack+first+order+resnet&client=ubuntu&hs=Es9&channel=fs&ei=3dwUYbahHcX-sAfluKDgCg&oq=backpack+first+order+resnet&gs_lcp=Cgdnd3Mtd2l6EAM6CwgAEIAEELEDEIMBOggIABCABBCxAzoRCC4QgAQQsQMQgwEQxwEQ0QM6DgguEIAEELEDEMcBEKMCOgsILhCABBDHARDRAzoECAAQQzoECC4QQzoOCC4QgAQQsQMQxwEQ0QM6CwguEIAEELEDEIMBOggILhCABBCxAzoHCC4QsQMQQzoLCC4QgAQQxwEQrwE6CAguELEDEIMBOgcIABCxAxBDOgwIABCxAxCDARAKEEM6BQgAEIAEOgQIABANOgoILhDHARCvARANOgcIABCxAxANOgQILhANOgUILhCABDoICC4QgAQQkwI6BggAEBYQHjoECAAQEzoICAAQFhAeEBM6CggAEBYQChAeEBM6CAgAEBYQChAeOggIABAIEA0QHjoICCEQFhAdEB46BwghEAoQoAE6BQghEKABSgQIQRgAUP3aBFiIgwVgvoQFaAFwAHgAgAFoiAH7D5IBBDI2LjKYAQCgAQHAAQE&sclient=gws-wiz&ved=0ahUKEwj2g4q5iavyAhVFP-wKHWUcCKwQ4dUDCA0&uact=5)

Reply:
I think it's impossible to figure out if this example is referenced nowhere. So we should assume it is reference, and add a message like
```
This tutorial has moved. Click [here](link to new tutorial) to continue to its new location.
```

Reply:
Reintroduced it. I hope the link works. When building the docs locally it just displays as plain text.